cdap-docs/tools/docs-extract-table.py

```diff
@@ -0,0 +1,163 @@
+    """
+
+    pieces = [piece.strip() for piece in line.split(PIPE)][1:-1]
+    pieces[0] = "``%s``" % pieces[0]
+    pieces = ["\"%s\"" % piece for piece in pieces]
+    comma = ','
+    return comma.join(pieces)
+
+def create_from_lines(line1, line2):
+    """ Combines two lines from a Markdown table and converts them into a single line for a reST CSV table line.
+    """
+    pieces1 = [piece.strip() for piece in line1.split(PIPE)][1:-1]
+    pieces2 = [piece.strip() for piece in line2.split(PIPE)][1:-1]
+    
+    if len(pieces1) != len(pieces2):
+        return ''
+        
+    pieces = []
+    for i in range(len(pieces1)):    
+        pieces.append(pieces1[i] + ' ' + pieces2[i])
+    
+    pieces = [piece.strip() for piece in pieces]
+    pieces[0] = "``%s``" % pieces[0]
+    pieces = ["\"%s\"" % piece for piece in pieces]
+    comma = ','
+    return comma.join(pieces)
+
+#
+# Create the table
+#
+
+TABLE_RULE = '+=='
+TABLE_TEXT = '| '
+TABLE_TEXT_CONTINUED = '|  '
+
+def create_table(input_file, output_file):
+    """ Creates the table by reading in a file, building the table, outputing to a new file
+    """
+    lines = []
+    return_code = 0
+    if os.path.isfile(input_file):
+        print "Reading in %s" % input_file
+        with open(input_file,'r') as f:
+            in_table = False
+            in_literal = False
+            skip_section = False
+            
+            raw_lines = []
+            
+            for line in f:
+                line = line.strip()
+                
+                if not in_table and line.startswith(TABLE_RULE):
+                    in_table = True
+                    in_header = True
+                    continue
+                if in_table and in_header and line.startswith(TABLE_TEXT):
+                    continue
+                if in_table and in_header and line.startswith(TABLE_RULE):
+                    in_header = False
+                    continue
+                if in_table and not in_header and line.startswith(TABLE_TEXT):
+                    if line.startswith(TABLE_TEXT_CONTINUED):
+                        # drop previous line
+                        # add to previous line
+                        lines.pop()
+                        previous_line = raw_lines.pop()
+                        new_line = create_from_lines(previous_line, line)
+                    else:
+                        raw_lines.append(line)
+                        new_line = create_parsed_line(line)
+                    lines.append(SPACES + new_line)
+                    continue
+                if in_table and not in_header and line.startswith(TABLE_RULE):
+                    in_table = False
+                    break                    
+    else:
+        print "Did not find %s" % input_file
+        print "Wrote 'missing file' to table"
+        lines.append(MISSING_FILE_TEMPLATE % input_file)
+        return_code = 2
+            
+    output = open(output_file,'w')
```

Comment:
line 141 is redundant. Not sure if opening the same file for writing would cause any weird issues.